src/intern_blks/ya_intern.c

```diff
@@ -227,16 +230,28 @@ void ya_int_song(ya_block_t *blk) {
 		if (strcmp("Playing", s) == 0) {
 			CHECK_PLAYER_ERROR(e_tmp);
 
-			sprintf(
-				startstr,
-				"%s %s %s",
-				playerctl_player_get_artist(player, &e_artist),
-				blk->internal->option[1],
-				playerctl_player_get_title(player, &e_title)
-			);
+			startstr[0] = 0;
+			attribute = strtok(ref, ";");
```

Comment:
Parsing ideally should be done before the while loop. Also, using spaces as separators would be preferred (see other blocks which use `sscanf()`).